cmd/opampsupervisor/supervisor/supervisor.go

```diff
@@ -1022,34 +1091,57 @@ func (s *Supervisor) composeOpAMPExtensionConfig() []byte {
 	return cfg.Bytes()
 }
 
-func (s *Supervisor) composeAgentConfigFiles() []byte {
-	conf := confmap.New()
+type configComposer func() []byte
+
+func (s *Supervisor) composeAgentConfigFiles(incomingConfig *protobufs.AgentRemoteConfig) ([]byte, error) {
+	konf := koanf.New("::")
```

Comment:
Could we call this `conf` or similar? `konf` surprised me at first when reading it because it reads like a misspelling of `koanf`.